packages/admin/src/form/FinalFormRangeSlider.tsx

```diff
@@ -0,0 +1,136 @@
+import { FormControl, Slider, SliderProps, Theme, WithStyles } from "@material-ui/core";
+import { createStyles, withStyles } from "@material-ui/styles";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+
+import { InputBase } from "./InputBase";
+
+export type CometAdminFinalFormRangeSliderClassKeys =
+    | "root"
+    | "inputsWrapper"
+    | "inputFieldsSeperatorContainer"
+    | "sliderWrapper"
+    | "inputFieldContainer";
+
+const styles = (theme: Theme) => {
+    return createStyles<CometAdminFinalFormRangeSliderClassKeys, any>({
+        root: {
+            padding: "0 20px",
+            width: "100%",
+        },
+        inputsWrapper: {
+            justifyContent: "space-between",
+            marginBottom: "15px",
+            alignItems: "center",
+            display: "flex",
+        },
+        inputFieldsSeperatorContainer: {
+            textAlign: "center",
+            minWidth: "20%",
+        },
+        sliderWrapper: {
+            paddingBottom: "20px",
+        },
+        inputFieldContainer: {
+            textAlign: "center",
+        },
+    });
+};
+
+interface IFinalFormRangeSliderProps extends FieldRenderProps<{ min: number; max: number }, HTMLInputElement> {
+    min: number;
+    max: number;
+    startAdornment?: string | React.ReactElement;
+    endAdornment?: string | React.ReactElement;
+    sliderProps?: Omit<SliderProps, "min" | "max">;
+}
+
+const FinalFormRangeSliderComponent: React.FunctionComponent<WithStyles<typeof styles, true> & IFinalFormRangeSliderProps> = ({
+    classes,
+    min,
+    max,
+    sliderProps,
+    startAdornment,
+    endAdornment,
+    input: { name, onChange, value: fieldValue },
+}) => {
+    const [internalMinInput, setInternalMinInput] = React.useState(fieldValue.min || 0);
+    const [internalMaxInput, setInternalMaxInput] = React.useState(fieldValue.max || 0);
+
+    const handleSliderChange = (event: React.ChangeEvent<HTMLInputElement>, newValue: number[]) => {
+        onChange({ min: newValue[0], max: newValue[1] });
+    };
+
+    React.useEffect(() => {
+        setInternalMinInput(fieldValue.min);
+        setInternalMaxInput(fieldValue.max);
+    }, [fieldValue.min, fieldValue.max]);
```

Comment:
```suggestion
    }, [fieldValue]);
```

Reply:
Testing for primitives in the deps-array of useEffect is more reliable than testing for objects. So I think the original version is better.

I made a small demonstration which shows that testing for an object may result in too many calls of the useEffect-callback https://codesandbox.io/s/purple-cloud-jp9si?file=/src/App.tsx

Reply:
Cool! Nice demo! 🙂